### ISSUE TITLE
unable to deploy privileged pod after 1.8 upgrade unless I set allowPrivilegeEscalation true

### ISSUE THREAD

**Issue by jhorwit2** [kubernetes/kubernetes#53437]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

I upgraded a cluster to 1.8 and ran into an issue with pods that use `privileged: true` and don't set `allowPrivilegeEscalation: true`. 

The PSP I had created and been using prior to 1.8:
```
apiVersion: extensions/v1beta1
kind: PodSecurityPolicy
metadata:
  name: privileged
spec:
  privileged: true
  hostNetwork: true
  hostPID: true
  hostIPC: true
  hostPorts:
  - min: 1
    max: 32000
  fsGroup:
    rule: RunAsAny
  runAsUser:
    rule: RunAsAny
  seLinux:
    rule: RunAsAny
  supplementalGroups:
    rule: RunAsAny
  volumes:
  - '*'
```

The error I got when applying an update to the canal daemonset:

```
  Warning  FailedCreate  10m                 daemon-set  Error creating: Pod "canal-z8dmj" is invalid: [spec.containers[0].securityContext: Invalid value: api.SecurityContext{Capabilities:(*api.Capabilities)(nil), Privileged:(*bool)(0xc424f9c2c0), SELinuxOptions:(*api.SELinuxOptions)(nil), RunAsUser:(*int64)(nil), RunAsNonRoot:(*bool)(nil), ReadOnlyRootFilesystem:(*bool)(nil), AllowPrivilegeEscalation:(*bool)(0xc427e10c38)}: cannot set `allowPrivilegeEscalation` to false and `privileged` to true, spec.containers[2].securityContext: Invalid value: api.SecurityContext{Capabilities:(*api.Capabilities)(nil), Privileged:(*bool)(0xc424f9c2ca), SELinuxOptions:(*api.SELinuxOptions)(nil), RunAsUser:(*int64)(nil), RunAsNonRoot:(*bool)(nil), ReadOnlyRootFilesystem:(*bool)(nil), AllowPrivilegeEscalation:(*bool)(0xc427e10c38)}: cannot set `allowPrivilegeEscalation` to false and `privileged` to true]
```

**What you expected to happen**:

I expect the pod to be valid since it was not noted as a breaking change in the release notes for 1.8.

**How to reproduce it (as minimally and precisely as possible)**:

Using the PSP above, attempt to create a pod with `privileged: true` and no value for `allowPrivilegeEscalation`. 

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.0", GitCommit:"6e937839ac04a38cac63e6a7a306c5d035fe7b0a", GitTreeState:"clean", BuildDate:"2017-09-28T22:57:57Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.0", GitCommit:"6e937839ac04a38cac63e6a7a306c5d035fe7b0a", GitTreeState:"clean", BuildDate:"2017-09-28T22:46:41Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

cc @liggitt @tallclair @jessfraz

**Comment by jhorwit2** [comment]
semi-unrelated: the error message on that event is hideous 👼

**Comment by liggitt** [comment]
@jessfraz @tallclair existing PSPs should continue to work as they did before. doesn't that mean `DefaultAllowPrivilegeEscalation` has to default to true in PSPs if unspecified?

**Comment by liggitt** [comment]
(alternatively, `psp.Spec.AllowPrivilegeEscalation` could have been a `*bool`, defaulting to nil/no-opinion, but that's not possible since it shipped as `bool` in 1.8.0)

**Comment by liggitt** [comment]
fixed in 1.8.1
